lib/active_model/serializer/adapter/json.rb

```diff
@@ -7,37 +7,16 @@ class Json < Adapter
         def serializable_hash(options = nil)
           options ||= {}
           if serializer.respond_to?(:each)
-            @result = serializer.map { |s| FlattenJson.new(s).serializable_hash(options) }
+            @result = serialize_array_without_root(serializer, options)
           else
-            @hash = {}
+            @result = resource_object_for(serializer)
+            @result = add_resource_relationships(
+              @result,
+              serializer,
+              serializer.options[:include])
```

Comment:
your right, so because in `add_resource_relationship` we *then* iterate over the associations and use the association options, this third parameter here isn't needed at all.